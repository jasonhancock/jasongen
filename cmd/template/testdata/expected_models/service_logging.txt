// Code generated by template.test. DO NOT EDIT.
// template.test 1.2.3

package widgets

import (
	"context"

	models "github.com/example/somemodels"
)

var _ SVC = (*LoggingService)(nil) // Verify that *LoggingService implements SVC.

type ErrorLogger interface {
	LogError(msg string, err error)
}

type LoggingService struct {
	logger ErrorLogger
	svc    SVC
}

func NewLoggingService(svc SVC, l ErrorLogger) *LoggingService {
	return &LoggingService{
		logger: l,
		svc:    svc,
	}
}

// WidgetCreate
func (s *LoggingService) WidgetCreate(ctx context.Context, req models.WidgetCreateRequest) (models.Widget, error) {
	resp, err := s.svc.WidgetCreate(ctx, req)
	if err != nil {
		s.logger.LogError("widgetCreate error", err)
	}

	return resp, err
}

// WidgetDelete delete a specific widget by ID.
func (s *LoggingService) WidgetDelete(ctx context.Context, id string) error {
	err := s.svc.WidgetDelete(ctx, id)
	if err != nil {
		s.logger.LogError("widgetDelete error", err)
	}

	return err
}

// WidgetGet get a specific widget by ID.
func (s *LoggingService) WidgetGet(ctx context.Context, id string, num int64) (models.Widget, error) {
	resp, err := s.svc.WidgetGet(ctx, id, num)
	if err != nil {
		s.logger.LogError("widgetGet error", err)
	}

	return resp, err
}

// WidgetsList gets a list of all widgets
func (s *LoggingService) WidgetsList(ctx context.Context, qp WidgetsListParams) (models.WidgetsListResponse, error) {
	resp, err := s.svc.WidgetsList(ctx, qp)
	if err != nil {
		s.logger.LogError("widgetsList error", err)
	}

	return resp, err
}
