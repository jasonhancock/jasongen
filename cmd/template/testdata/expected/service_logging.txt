// Code generated by template.test. DO NOT EDIT.
// template.test 1.2.3

package widgets

import "context"

var _ SVC = (*LoggingService)(nil) // Verify that *LoggingService implements SVC.

type ErrorLogger interface {
	LogError(msg string, err error, keyvals ...any)
}

type LoggingService struct {
	logger ErrorLogger
	svc    SVC
}

func NewLoggingService(svc SVC, l ErrorLogger) *LoggingService {
	return &LoggingService{
		logger: l,
		svc:    svc,
	}
}

// WidgetCreate
func (s *LoggingService) WidgetCreate(ctx context.Context, req WidgetCreateRequest) (Widget, error) {
	resp, err := s.svc.WidgetCreate(ctx, req)
	if err != nil {
		s.logger.LogError("widgetCreate error", err)
	}

	return resp, err
}

// WidgetDelete delete a specific widget by ID.
func (s *LoggingService) WidgetDelete(ctx context.Context, id string) error {
	err := s.svc.WidgetDelete(ctx, id)
	if err != nil {
		s.logger.LogError("widgetDelete error", err)
	}

	return err
}

// WidgetGet get a specific widget by ID.
func (s *LoggingService) WidgetGet(ctx context.Context, id string, num int64) (Widget, error) {
	resp, err := s.svc.WidgetGet(ctx, id, num)
	if err != nil {
		s.logger.LogError("widgetGet error", err)
	}

	return resp, err
}

// WidgetsList gets a list of all widgets
func (s *LoggingService) WidgetsList(ctx context.Context, qp WidgetsListParams) (WidgetsListResponse, error) {
	resp, err := s.svc.WidgetsList(ctx, qp)
	if err != nil {
		s.logger.LogError("widgetsList error", err)
	}

	return resp, err
}
